wallet/src/wallet/mod.rs

```diff
@@ -330,7 +330,10 @@ impl<B: storage::Backend> Wallet<B> {
     /// Returns true if a migration needs to be ran, false if it is already on the latest version
     /// and an error if it is an unsupported version
     pub fn check_db_needs_migration(db: &Store<B>) -> WalletResult<bool> {
-        match db.get_storage_version()? {
+        let db_tx = db.transaction_ro()?;
+        let version_result = db_tx.get_storage_version()?;
+        drop(db_tx);
```

Comment:
Could the `db_tx` be made temporary to avoid the setup/drop code overhead?
```suggestion
        let version_result = db.transaction_ro()?.get_storage_version()?;
```